### ISSUE TITLE
HTML minification incorrectly identifies "0" as empty attribute value

### ISSUE THREAD

**Issue by westonruter** [W3EDGE/w3-total-cache#90]
HTML like this:

```html
<iframe style="border: none; overflow: hidden;" src="https://www.facebook.com/plugins/post.php?href=https%3A%2F%2Fwww.facebook.com%2FTheRadfordFamily%2Fphotos%2Fa.506512802742080.1073741827.464409336952427%2F1686097548116927%2F%3Ftype%3D3&amp;width=500" width="500" height="479" frameborder="0" scrolling="no"></iframe>
```

Is erroneously gettings its HTML minified with `frameborder="0"` being replaced with just `frameborder`, as if it was a boolean attribute. This is not correct. It should be getting minified rather as `frameborder=0`.

This has been reported twice for the AMP plugin:

* https://github.com/ampproject/amp-wp/issues/3882
* https://wordpress.org/support/topic/amp-validation-issues/

**Comment by mavas84** [comment]
Hello @westonruter,
Thank you for bringing this to our attention. Can you please apply this [fix](https://github.com/W3EDGE/w3-total-cache/commit/f0faeb5cc3e2b807f153327c62a8d3c365e2757a#diff-9cea5333909e4ce25c7c3a5056c00737R356): 
and replace line 356 in  lib/Minify/Minify/HTML.php
`if ( empty( $m[3] ) ) {`
with
`if ( $m[3] === '' ) {`
Please let us know if this helps!

**Comment by westonruter** [comment]
@Mte90 would you please try that patch? I don't use the plugin myself.

**Comment by Mte90** [comment]
I cannot do a test right now, I have to wait the website that had issues with the amp validation and we cannot invest more time on that.
Looking at the code anyway seems that should work.

**Comment by maxicus** [comment]
Thanks for informing. Yes, it's a v0.11 regression.
Fixed already by https://github.com/W3EDGE/w3-total-cache/commit/f0faeb5cc3e2b807f153327c62a8d3c365e2757a will come out in a next release.

**Comment by djixas** [comment]
> Thanks for informing. Yes, it's v0.11 regression.
> Fixed already by [f0faeb5](https://github.com/W3EDGE/w3-total-cache/commit/f0faeb5cc3e2b807f153327c62a8d3c365e2757a) will come out in a next release.

Here it says that after fix it did not work still

I am not upgrading to new version anymore, your plugin breaks most of the stuff i've seen

https://wordpress.org/support/topic/amp-validation-issues/
